nixos/modules/services/mail/maddy.nix

```
@@ -215,6 +214,65 @@ in {
         '';
       };
 
+      tls = {
+        loader = mkOption {
+          type = with types; nullOr (enum [ "file" "off" ]);
+          default = "off";
+          description = lib.mdDoc ''
+            TLS certificates are obtained by modules called "certificate
+            loaders". Currently only the file loader is supported which reads
+            certificates from files specifying the options `keyPaths` and
+            `certPaths`.
+          '';
+        };
+
+        certificates = mkOption {
+          type = with types; nullOr (listOf (submodule {
+            options = {
+              keyPath = mkOption {
+                type = with types; nullOr path;
+                example = "/etc/ssl/mx1.example.org.key";
+                default = null;
+                description = lib.mdDoc ''
+                  Path to the private key used for TLS.
+                '';
+              };
```

Why allow null here? If any attrset of this kind exists in the list, it requires both options to be set. I think it'd be best to represent this in the options instead of adding assertions on top.

```suggestion
                type = types.path;
                example = "/etc/ssl/mx1.example.org.key";
                description = lib.mdDoc ''
                  Path to the private key used for TLS.
                '';
              };
```